solidity_contracts/src/CairoPrecompiles/MulticallCairoCounterCaller.sol

```diff
@@ -0,0 +1,195 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0 <0.9.0;
+
+/// @notice A contract that performs various call types to the Kakarot MulticallCairo precompile.
+/// @dev Only meant to test the MulticallCairo precompile when called from a Solidity Contract.
+contract MulticallCairoCounterCaller {
+    using MulticallCairoLib for uint256;
+
+    /// @dev The cairo contract to call
+    uint256 immutable cairoCounter;
+
+    /// @dev The cairo function selector to call - `inc`
+    uint256 constant FUNCTION_SELECTOR_INC = uint256(keccak256("inc")) % 2 ** 250;
+
+    /// @dev The cairo function selector to call - `set_counter`
+    uint256 constant FUNCTION_SELECTOR_SET_COUNTER = uint256(keccak256("set_counter")) % 2 ** 250;
+
+    constructor(uint256 cairoContractAddress) {
+        cairoCounter = cairoContractAddress;
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter
+    function incrementCairoCounter() external {
+        cairoCounter.callCairo("inc");
+    }
+
+    /// @notice Calls the Cairo contract to increment its internal counter in a batch of multiple calls
+    function incrementCairoCounterBatch(uint32 n_calls) external {
+        uint256[][] memory calls = new uint256[][](n_calls);
+        for (uint32 i = 0; i < n_calls; i++) {
+            calls[i] = new uint256[](3);
+            calls[i][0] = cairoCounter;
+            calls[i][1] = FUNCTION_SELECTOR_INC;
+            calls[i][2] = 0; // Length of the empty calldata array
```

Comment:
can't we create a solidity `struct` representing a cairo call (and consequently have this `0` automatically put by the compiler)